Reduce click-area of mazemap checkbox label

norbye opened the pull request:
# Description

Reduce click-area and margin of mazemap checkbox label as reported in https://github.com/webkom/lego/issues/3567.

Also reduces the click area of all checkboxes that are wrapped in the Field component across the webapp.

# Result

If you've made visual changes, please check the boxes below and include images showing the changes. Descriptions are appreciated.

- [x] Changes look good on both light and dark theme.
- [x] Changes look good with different viewports (mobile, tablet, etc.).
- [x] Changes look good with slower Internet connections.

> [!CAUTION]
> Make sure your images do not contain any real user information.

<table>
    <tr>
        <td width="25%">Description</td>
        <td>Before</td>
        <td>After</td>
    </tr>
    <tr>
        <td>
            Reduced size and margin
        </td>
        <td><img width="478" alt="image" src="https://github.com/webkom/lego-webapp/assets/13599770/32014ee4-f54c-46e0-98eb-de4327867356"></td>
        <td><img width="371" alt="image" src="https://github.com/webkom/lego-webapp/assets/13599770/6d0d3f66-abbd-406c-9b57-30b8f21f900a">
        </td>
    </tr>
</table>

# Testing

- [x] I have thoroughly tested my changes.

Please describe what and how the changes have been tested, and provide instructions to reproduce if necessary.

---

Resolves https://github.com/webkom/lego/issues/3567

ivarnakken commented:
Is this a general problem with all checkboxes in forms or just this specific one?

norbye commented:
> Is this a general problem with all checkboxes in forms or just this specific one?

Hmm good point - I figured we could just leave it as is everywhere as I've never seen it to be a problem - but it does seem like the more intuitive way to make it fit-content for all labels across the webapp

ShaileshS1702 reviewed (approved):
LGTM!

eikhr reviewed (approved):
This happens with all checkboxes in the webapp, but I don't think it's as much of a problem in forms.

norbye commented:
I agree that it's not the biggest of problems across the webapp - but it isn't really ideal either, soo I changed the PR.

Updated it to fit-content for all labels that are wrapped in the `Field` component. Have tested a few occurrences and they look good (generally this shouldn't influence any styling, as all the label components that are affected is wrapped with a flex component that has not been modified)
